config/initializers/form_builder.rb

```diff
@@ -0,0 +1,22 @@
+class ActionView::Helpers::FormBuilder
+  alias :orig_label :label
+
+# add a '*' after the field label if the field is required
+# see http://davidsulc.com/blog/2011/05/01/self-marking-required-fields-in-rails-3/
+# Also http://blog.pothoven.net/2012/10/self-marking-required-fields-in-rails.html
+  def label(method, content_or_options = nil, options = nil, &block)
+    if content_or_options && content_or_options.class == Hash
+      options = content_or_options
+    else
+      content = content_or_options
+    end
+
+    required_mark = ''
+    required_mark = ' *'.html_safe if object.class.validators_on(method).map(&:class).include? ActiveRecord::Validations::PresenceValidator
```

Comment:
This doesn't totally work for Partners. In particular for the Province and Status fields, but the form controls for those fields don't allow a blank setting anyway (at least not in the hq implementation). 
The label functionality works by checking if the field passed in has got a validator presence: true in the model.
For Province it doesn't work because the form field is province_id, and in the model it has
validates :province, presence: true
For Status, the model doesn't have any associated presence: true.
One solution could be to set a CSS class of "required" against a field, instead of adding the asterisk, and then use CSS to add the asterisk. With that way you could set the CSS "required" class manually in the form against anything else like Province or Status. 
Thoughts?